### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by ayr-ton** [PyratLabs/ansible-role-k3s#176] [Ideas]
Motivated on the discussion started here:
https://github.com/k3s-io/k3s-ansible/issues/39#issuecomment-1008061739

Which direction do you folks see this repository taking for the future?

Do you have suggestions or bugs not reported as issues yet? 

Have a PR in mind and want to discuss if it make sense as an issue? 

Let's burst some contributions from this async discussion. :-)

**Comment by jon-stumpf** [comment]
I still need to finish my review of **ansible-role-k3s**.  I am very familiar with **k3s-ansible**.

In **k3s-ansible**, I have submitted three, cumulative PRs but received no feedback from the maintainers:

- k3s-io/k3s-ansible#158: made the _reset_ playbook be at parity with [https://get.k3s.io](https://get.k3s.io) _k3s-uninstall.sh_
- k3s-io/k3s-ansible#164: implemented most of the install switches from [https://get.k3s.io](https://get.k3s.io) 
- k3s-io/k3s-ansible#166: finished HA

In reviewing the issues and PRs, the two things that I would like to see implemented are:

1. k3s-io/k3s-ansible#52
2. k3s-io/k3s-ansible#114

The rest I believe can be closed/cancelled (or I have implemented it already).

For **ansible-role-k3s**, I first intend to replace _k3s-uninstall.sh_ and _k3s-killall.sh_ with ansible equivalents (done in jon-stumpf/k3s-ansible).  That will be my first pull request.

And, for the merger, I would like to get **k3s-io** to agree on hosting this repo.

**Comment by utkuozdemir** [comment]
A few things come to my mind:

- Testing upgrades from CI - I had a problem on upgrading from k3s `v1.22.3+k3s1` to `v1.22.4+k3s1` with this module. It was related to the certificates under `/var/lib/rancher/k3s/server/tls`. Didn't dig deep into it to give more details (since I stopped using it afterwards), but some additional testing can help here.

- Make the role do things more similar to other tools:  I think it'd be beneficial to make the end state more similar to what k3s's own `install.sh` or `k3sup` does - same files at same locations with same content. Doing it might reduce the chances of something going wrong (example: maybe remove the soft-links hard-links thing)

- I had a problem with `free` strategy with nodes not respecting `--node-ip` and `--node-external-ip` args correctly. Again, I didn't dig deeper. If I have some time I can look into reproducing it and opening an issue. But overall, making sure that role works fine with `free` strategy can be a good idea (or if it doesn't work with `free`, document it and/or fail on preflight).
